### PR TITLE
[AKS] `az aks update`: Add parameter `--network-plugin` to Update command to support migrating to Azure CNI Overlay

### PR DESCRIPTION
**Pull request by tyler-lloyd**
**Related command**

`az aks update`

**Description**

Allow aks clusters to update their `network-plugin` property from the cli. This allows clusters using kubenet
to upgrade to CNI Overlay. See https://learn.microsoft.com/en-us/azure/aks/azure-cni-overlay?tabs=kubectl#upgrade-an-existing-cluster-to-cni-overlay for more information.

**Testing Guide**

1. Create an aks cluster with kubenet - `az aks create -n mycluster -g myrg --network-plugin kubenet`
2. Update the cluster to migrate from kubenet to cni overlay - `az aks update -n mycluster -g myrg --network-plugin azure --network-plugin-mode overlay`

**History Notes**

[AKS] `az aks update`: Add parameter `--network-plugin` to update the network plugin of the AKS cluster

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Comment by tyler-lloyd** [comment]
PTAL @zhoxing-ms @yanzhudd

**Comment by tyler-lloyd** [comment]
FYI @FumingZhang

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -7896,6 +7896,48 @@ def test_aks_azure_cni_overlay_migration(self, resource_group, resource_group_lo
         self.cmd(
             'aks delete -g {resource_group} -n {name} --yes --no-wait', checks=[self.is_empty()])
 
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='eastus2', preserve_default_location=True)
+    def test_aks_kubenet_to_cni_overlay_migration(self, resource_group, resource_group_location):
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:7901]: Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=84781001&view=results) to validate the change.

File: src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py [hunk 2]
```diff
@@ -2128,13 +2128,13 @@ def _get_network_plugin_mode(self, enable_validation: bool = False) -> Union[str
         # try to read the property value corresponding to the parameter from the `mc` object
 
         # if create, try and read from the mc object
```
[the remark this answers is not in the document]
Comment (tyler-lloyd) [reply src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py]: removed

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py (hunk 1 above)
Comment (tyler-lloyd) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:7901]: looks green to me

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@FumingZhang Could you please review this PR again?

**Review by FumingZhang — APPROVED** [review]
LGTM

**Pull request merged**